### ISSUE TITLE
k8up doesn't cleanup PreBackup Pod when backup is finished

### ISSUE THREAD

**Issue by TheBigLee** [vshn/k8up#413]
## Describe the bug

we noticed a lot of prebackuppod running only with `sleep infinity` (so an increase of the cluster pod count up to 2492)

example of our current prebackup template

```yaml
spec:                                                                                                                                                                         
  backupCommand: /bin/sh -c "dump=$(mktemp) && mysqldump --max-allowed-packet=500M                                                                                            
    --events --routines --quick --add-locks --no-autocommit --single-transaction --no-create-db                                                                               
    --no-data -h $BACKUP_DB_HOST -u $BACKUP_DB_USERNAME -p$BACKUP_DB_PASSWORD $BACKUP_DB_DATABASE                                                                             
    > $dump && mysqldump --max-allowed-packet=500M --events --routines --quick --add-locks                                                                                    
    --no-autocommit --single-transaction --no-create-db --ignore-table=$BACKUP_DB_DATABASE.watchdog                                                                           
    --no-create-info -h $BACKUP_DB_HOST -u $BACKUP_DB_USERNAME -p$BACKUP_DB_PASSWORD                                                                                          
    $BACKUP_DB_DATABASE >> $dump && cat $dump && rm $dump"                                                                                                                    
  fileExtension: .mariadb.sql                                                                                                                                                 
  pod:                                                                                                                                                                        
    metadata:                                                                                                                                                                 
      labels:                                                                                                                                                                 
        app.kubernetes.io/instance: mariadb                                                                                                                                   
        app.kubernetes.io/managed-by: Helm                                                                                                                                    
        app.kubernetes.io/name: mariadb-dbaas
        helm.sh/chart: mariadb-dbaas-0.1.0
        <redacted>
        prebackuppod: mariadb
    spec:
      containers:
      - args:
        - sleep
        - infinity
        env:
```

we've created our template from https://k8up.io/k8up/1.0.0/how-tos/prebackuppod.html

```console
k top pod --sort-by=cpu -A | grep backup
namespaceW                              mariadb-prebackuppod-5ddc7b9888-zbcf2                               0m           0Mi             
namespaceX                               mariadb-prebackuppod-7bc8c87df8-66fm4                               0m           0Mi             
namespaceY                               mariadb-prebackuppod-7bc8c87df8-qpssv                               0m           0Mi             
namespaceZ                               mariadb-prebackuppod-56f649749c-qss9t                               0m           0Mi
```

```console
namespaceFoo mariadb-prebackuppod-7bc8c87df8-t4td2 -- bash 
PID   USER     TIME  COMMAND
    1 10006000  0:00 sleep infinity
   19 10006000  0:00 bash
   26 10006000  0:00 ps aux
```
Are these pods be tracked in any way by k8up and removed once the backup is done?

## Expected behavior

Prebackup pods are terminated and cleaned up properly.

## Environment (please complete the following information):

- Image Version: v1.0.3
- K8s Distribution: OpenShift

**Comment by cimnine** [comment]
As a work-around, you may try to set (or reduce) the `keepJobs` property [on your `Schedule` resource](https://k8up.io/k8up/1.0.0/references/object-specifications.html#_schedule) or [define the `BACKUP_GLOBALKEEPJOBS` environment variable](https://k8up.io/k8up/1.0.0/references/config-reference.html) for the operator.

**Comment by cdchris12** [comment]
> As a work-around, you may try to set (or reduce) the `keepJobs` property [on your `Schedule` resource](https://k8up.io/k8up/1.0.0/references/object-specifications.html#_schedule) or [define the `BACKUP_GLOBALKEEPJOBS` environment variable](https://k8up.io/k8up/1.0.0/references/config-reference.html) for the operator.
> 
> This should work, because the PreBackupPod Deployment is owned (via `ownerReference`) by the corresponding `Backup` instance.

We have tested this on our `3.11` OpenShift with version `v1.0.1` of the k8up operator, and both of these solutions did not resolve the issue for us initially. It seems that our k8up operator was configured with a maximum RAM limit of 256MB, which the pod hit reliably at the end of every completed job. We've quadrupled that memory limitation to 1GB, and the suggested fix now appears to work as intended for our cluster.

We'll report back again after tomorrow's backup period with more information on whether or not this fully resolves our issues.

**Comment by cimnine** [comment]
I believe I've identified the problem in the meantime. A fix is now being discussed in #414.

**Comment by cimnine** [comment]
The proposed fix was merged and [a new release, `v1.0.4`, is ready](https://github.com/vshn/k8up/releases/tag/v1.0.4).
The helm chart [will be updated soonish™️](https://github.com/appuio/charts/pull/281) as well.

Please let us know whether this fixed the reported misbehavior.
